Browser_IDE/IDBStoredProject.js

```diff
@@ -330,34 +337,41 @@ class __IDBStoredProjectRW{
     request(transaction, func)
     {
         return new Promise((resolve, reject) => {
-            let result = func();
-            result.onerror = function(){console.log("error");transaction.abort(); reject(result.error);};
-            result.onsuccess = function(){
-            resolve(result.result);};
+            let result = undefined;
+            
+            try {
+                result = func();
+            } catch(err){
+                reject(err);
+                return;
+            }
+
+            result.onerror = function(){console.log("error");transaction.abort(); reject(transaction.error);};
```

Comment:
Should this be `result.error` rather than `transaction.error`? I could be wrong, but I think the error is accessible from the IDBRequest object in this case, though I've been unable to test causing an error here so not really sure.